packages/cli/internal/pkg/osutils/utils_test.go

```diff
@@ -0,0 +1,58 @@
+package osutils
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/aws/amazon-genomics-cli/internal/pkg/cli/clierror/actionableerror"
+	iomocks "github.com/aws/amazon-genomics-cli/internal/pkg/mocks/io"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDetermineHomeDir_Success(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+	expectedPath := "/some/dir"
+	mockOs.EXPECT().UserHomeDir().Return(expectedPath, nil)
+	actualPath, err := DetermineHomeDir()
+
+	assert.NoError(t, err)
+	assert.Equal(t, expectedPath, actualPath)
+}
+
+func TestDetermineHomeDir_Failure(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+	expectedOsError := errors.New("some error")
+	mockOs.EXPECT().UserHomeDir().Return("", expectedOsError)
+	_, err := DetermineHomeDir()
+
+	expectedError := actionableerror.New(err, "Please check that your home or user profile directory is defined within your environment variables")
+
+	assert.Error(t, err, expectedError)
+}
+
+func TestExpandHomeDir_WithExpansion(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+
+	expectedHomePath := "home/user"
+	expectedExpandedPath := "home/user/FooBar"
+
+	mockOs.EXPECT().UserHomeDir().Return(expectedHomePath, nil)
+	actualExpandedPath, _ := ExpandHomeDir("~/FooBar")
+
+	assert.Equal(t, expectedExpandedPath, actualExpandedPath)
+	ctrl.Finish()
+}
+
+func TestExpandHomeDir_WithoutExpansion(t *testing.T) {
```

Comment:
1. I think for testing this function it would be great to employ parameterized tests (more info on that: https://gobyexample.com/testing) We have a bunch of good examples as well: (https://github.com/aws/amazon-genomics-cli/blob/119c6303ecfec6b6ac346e3fcdc22a364abba533/packages/cli/internal/pkg/cli/describe_output_test.go)
2. Please add more test cases  for edge cases (e.g. `~` in random places of the path, one can create an actual folder with this name. `\~\~\~/`,  `~\~\~`, `foo~/~bar`, `~foo/bar~`, etc)
3. Please add a _separate_ test for error handling.

Reply:
One more test case to add `~`.

Reply:
Just an empty string as an input is another good test case

Reply:
Added additional unit tests to cover the edge cases. Also utilized parameterized tests for a portion of the unit tests as suggested.